framework/src/systems/NonlinearSystemBase.C

```diff
@@ -294,7 +294,10 @@ NonlinearSystemBase::initialSetup()
         _fe_problem.getMortarInterfaces(/*displaced=*/false);
     for (const auto & mortar_interface : undisplaced_mortar_interfaces)
     {
-      auto primary_secondary_boundary_pair = mortar_interface.first;
+      const auto primary_secondary_boundary_pair = mortar_interface.first;
+      if (!_constraints.hasActiveMortarConstraints(primary_secondary_boundary_pair, false))
+        continue;
```

Comment:
@recuero this is the logic change that removes the failed assertions

Reply:
Got it. Whether or not we have mortar aux kernels doesn't really matter for this code. Here, only constraints are dealt with.